### PR TITLE
chore: update performance test to use failover mode

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

Performance tests were using the "readonly" flag to determine whether the driver should reconnect to the writer or connect to a reader during failover.
This approach is outdated after the failoverMode connection parameter is introduced, so tests will wait for TASK A to fail instead of finishing failover as soon as a reader connection is established.

### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by crystall-bitquill — APPROVED** [review]

File: wrapper/src/test/java/integration/container/tests/PerformanceTest.java [hunk 1]
```diff
@@ -279,9 +281,10 @@ private void execute_FailoverTime_SocketTimeout(int socketTimeout, int sleepDela
                 .getProxyDatabaseInfo()
                 .getInstanceEndpointSuffix());
     props.setProperty("failoverTimeoutMs", Integer.toString(PERF_FAILOVER_TIMEOUT_MS));
+    FailoverConnectionPlugin.FAILOVER_MODE.set(props, "strict-reader");
```
Comment (brunos-bq) [inline wrapper/src/test/java/integration/container/tests/PerformanceTest.java:284]: this is equivalent to setting that parameter to true?

File: wrapper/src/test/java/integration/container/tests/PerformanceTest.java [hunk 2]
```diff
@@ -163,7 +164,7 @@ private void execute_FailureDetectionTime_EnhancedMonitoringEnabled(
     PLUGINS.set(props, "efm");
 
     final PerfStatMonitoring data = new PerfStatMonitoring();
-    doMeasurePerformance(sleepDelayMillis, REPEAT_TIMES, props, false, data);
+    doMeasurePerformance(sleepDelayMillis, REPEAT_TIMES, props, data);
```
Comment (brunos-bq) [inline wrapper/src/test/java/integration/container/tests/PerformanceTest.java:167]: does that mean that what was false is now true?

**Review by sergiyvamz — APPROVED** [review]

File: wrapper/src/test/java/integration/container/tests/PerformanceTest.java (hunk 2 above)
Comment (crystall-bitquill) [reply wrapper/src/test/java/integration/container/tests/PerformanceTest.java:167]: No, the parameter was removed altogether from the doMeasurePerformance method and the failover mode does not default to being read onlt.

File: wrapper/src/test/java/integration/container/tests/PerformanceTest.java (hunk 1 above)
Comment (crystall-bitquill) [reply wrapper/src/test/java/integration/container/tests/PerformanceTest.java:284]: Yes

**Pull request merged**